pkg/controller/sparkapplication/controller.go

```diff
@@ -494,16 +494,21 @@ func (c *Controller) syncSparkApplication(key string) error {
 			appToUpdate = c.submitSparkApplication(appToUpdate)
 		}
 	case v1beta1.InvalidatingState:
+		if err := c.getAndUpdateAppState(appToUpdate); err != nil {
+			return err
+		}
+		glog.V(2).Infof("Invalidating app pod: %s", appToUpdate.Status.DriverInfo.PodName)
 		// Invalidate the current run and enqueue the SparkApplication for re-execution.
 		if err := c.deleteSparkResources(appToUpdate); err != nil {
 			glog.Errorf("failed to delete resources associated with SparkApplication %s/%s: %v",
 				appToUpdate.Namespace, appToUpdate.Name, err)
 			return err
 		}
-		c.clearStatus(&appToUpdate.Status)
```

Comment:
`clearStatus` resets the driverInfo, which the driver pod name is a part of. In order for `if c.validateSparkResourceDeletion(appToUpdate)` to do its job, the status should be reset only after the check is complete.

Reply:
`clearStatus` does not clear `DriverInfo` in `Invalidating` state. We should keep this call.

Reply:
You are right. I mis-read the code. Now added back.